src/store/auth/sagas.js

```diff
@@ -0,0 +1,86 @@
+import { takeLatest, call, put } from "redux-saga/effects";
+import { push } from "connected-react-router";
+import {
+  fetchAuthenticatedUser,
+  fetchAuthenticatedUserSuccess,
+  logoutSuccess,
+  loginSuccess,
+  loginError,
+  setToken,
+  forgotPasswordSuccess,
+  forgotPasswordError,
+  registerSuccess,
+  registerError,
+  resetPasswordSuccess,
+  resetPasswordError,
+  socialAuthSuccess,
+  socialAuthError,
+} from "./actions";
+import {
+  LOGIN_REQUEST,
+  FETCH_AUTHENTICATED_USER_REQUEST,
+  LOGOUT_REQUEST,
+  REGISTER_REQUEST,
+} from "./actionTypes";
+import { DASHBOARD, LOGIN } from "../../routes";
+import authService from "../../services/AuthService";
+import { removeItem } from "../../utils/localStorage";
+
+export function* authorize({ email, password }) {
+  try {
+    const token = yield call(authService.login, { email, password });
+    yield put(loginSuccess());
+    yield put(setToken(token));
+    yield put(push(DASHBOARD));
+  } catch (error) {
+    yield put(loginError());
+  }
+}
+
+export function* register({
+  firstName: first_name,
+  lastName: last_name,
+  username,
+  email,
+  password,
+}) {
+  try {
+    const token = yield call(authService.register, {
+      first_name,
+      last_name,
+      email,
+      password,
+      username,
+    });
+    yield put(setToken(token));
+    yield put(registerSuccess());
+    yield put(push(LOGIN));
+  } catch (error) {
+    yield put(registerError());
+  }
+}
+
+export function* logout() {
+  try {
+    yield put(logoutSuccess());
+    yield put(removeItem("token"));
+  } catch (error) {
+    //
```

Comment:
Create action for `LOGOUT_ERROR` even if you don't handle anything at this time.